.travis.yml

```diff
@@ -14,53 +15,89 @@ env:
     - BASEPATH: "$TRAVIS_BUILD_DIR"
     - ROOTSYS: "$BASEPATH/root"
 
+addons:
+  homebrew:
+    packages:
+      - boost
+      - python3
+      - pip3
+  apt:
+    update: true
+    packages:
+      - libboost-all-dev
+      - python3
+      - python3-pip
+      - python3-venv
+      - pandoc
+
+before_install:
+  - cd $BASEPATH
+  - python3 -m venv venv
+  - source ./venv/bin/activate
+
+install:
+  - python3 -m pip install scikit-build
+  - python3 -m pip install pytest==5.4.1
+  - python3 -m pip install pytest-cov
+  - python3 -m pip install scipy
+  - wget https://root.cern.ch/download/root_${ROOTBIN}.tar.gz
+  - tar xpvfz root_*.tar.gz > /dev/null 2>&1
+  - source root/bin/thisroot.sh
+  - cd $BASEPATH
+  - python3 setup.py  --generator "Unix Makefiles" --skip-generator-test install -- -- -j2
```

Comment:
The install step is meant for installing dependencies only. https://docs.travis-ci.com/user/job-lifecycle/#the-job-lifecycle
script should test if your code is building and the tests are running (travis should consider splitting script into build and test though)

Reply:
Yes I read that page too, but found that there are no such regulations there. So I assume it's mostly the name of those stages that suggests how to use it.

The reason I put the build under install is that the job itself is either testing or building documentation. It also allowed me to save lines.

Reply:
I like it that the relevant code to the project lives inside the script part. So putting the install process into install makes it slightly harder to orient yourself. I agree that the whole job lifecycle is not perfect in travis. But apart from a few lines of code you save, it just makes things harder to understand and less flexible.

Reply:
https://docs.travis-ci.com/user/build-stages/#how-to-define-build-stages
I think if we define our own two stages one for building and one for testing, that would be the way to go

Reply:
Oh yes that's a better solution. Sorry already ran this one https://github.com/ComPWA/pycompwa/pull/80/commits/3c2a35eb4c3756b3d869f36ef826b49509de2232